### ISSUE TITLE
How to handle packages requiring jQuery >= 3.x?

### ISSUE THREAD

**Issue by MacGyer** [yiisoft/yii2#14338]
Hi all,

I am using Yii2 2.0.12 with Composer and fxp-plugin to handle Bower/NPM assets.
In my composer.json I required a package, which itself requires jQuery >= 3.2.1 via Bower. 
After running `composer update`, it emits an error of not being able to determine correct jQuery version.

**As jQuery 2.x has reached EOL (see: [https://github.com/jquery/jquery.com/issues/162](https://github.com/jquery/jquery.com/issues/162)) what about jQuery 3 support?**

**How to handle packages which are requiring jQuery >= 3.x?**

### What steps will reproduce the problem?
require `"bower-asset/materialize": "0.99.0"` in composer.json or simply require `"bower-asset/jquery": "~3.2.1"` in composer.json.
Run `composer update`.

### What is the expected result?
Install jquery 3.2.1.

### What do you get instead?
Dependency resolving error.

### Additional info
Composer output:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - yiisoft/yii2 2.0.9 requires bower-asset/jquery.inputmask ~3.2.2 -> no matching package found.
    - yiisoft/yii2 2.0.8 requires bower-asset/jquery.inputmask ~3.2.2 -> no matching package found.
    - yiisoft/yii2 2.0.7 requires bower-asset/jquery.inputmask ~3.2.2 -> no matching package found.
    - yiisoft/yii2 2.0.6 requires bower-asset/jquery 2.1.*@stable | 1.11.*@stable -> no matching package found.
    - yiisoft/yii2 2.0.10 requires bower-asset/jquery.inputmask ~3.2.2 -> no matching package found.
    - yiisoft/yii2 2.0.12 requires bower-asset/jquery 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable -> satisfiable by bower-asset/jquery[2.2.4, 1.12.1-patch1].
    - yiisoft/yii2 2.0.11 requires bower-asset/jquery 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable -> satisfiable by bower-asset/jquery[2.2.4, 1.12.1-patch1].
    - yiisoft/yii2 2.0.11.1 requires bower-asset/jquery 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable -> satisfiable by bower-asset/jquery[2.2.4, 1.12.1-patch1].
    - yiisoft/yii2 2.0.11.2 requires bower-asset/jquery 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable -> satisfiable by bower-asset/jquery[2.2.4, 1.12.1-patch1].
    - yiisoft/yii2 2.0.12 requires bower-asset/jquery 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable -> satisfiable by bower-asset/jquery[2.2.4, 1.12.1-patch1].
    - Can only install one of: bower-asset/jquery[3.2.1, 2.2.4].
    - Can only install one of: bower-asset/jquery[1.12.1-patch1, 3.2.1].
    - bower-asset/materialize v0.99.0 requires bower-asset/jquery >=3.2.1 -> satisfiable by bower-asset/jquery[3.2.1].
    - Installation request for bower-asset/materialize 0.99.0 -> satisfiable by bower-asset/materialize[v0.99.0].
    - Installation request for yiisoft/yii2 ~2.0.6 -> satisfiable by yiisoft/yii2[2.0.10, 2.0.11, 2.0.11.1, 2.0.11.2, 2.0.12, 2.0.6, 2.0.7, 2.0.8, 2.0.9].
```

composer.json
```
"require": {
        "php": ">=5.4.0",
        "yiisoft/yii2": "~2.0.6",
        "yiisoft/yii2-swiftmailer": "~2.0.0",
        "bower-asset/materialize": "0.99.0"
    },

```

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 7.1.6
| Operating system | Ubuntu 16.04

**Comment by MacGyer** [comment]
@samdark Sounds good! Thanks

**Comment by SilverFire** [comment]
Checked jQuery upgrade guide from 2.* to 3.0, fixed incompatibilities (70ad9f730b8cef64f7243895f737f0df856469bc, 3a8feb17a8213362d79bfbb32dc89621c1de42f1), checked basic application and allowed existing 3.* versions in `composer.json`: 3a8feb17a8213362d79bfbb32dc89621c1de42f1

**Comment by cebe** [comment]
@SilverFire I fixed composer.json in `framework/` for you ee74257 ;)
